### PR TITLE
Added flavors for OCP-4.17 Baremetal, ESX and Openstack

### PR DESCRIPTION
**Pull request by burhan20**

### CODE REVIEW

**Comment by burhan20** [comment]
> Thanks @burhan20 , please hide 4.14 flavors and also make the corresponding changes on the aci-containers side.

Sure, Sumit

File: provision/testdata/list_flavors.stdout.txt [hunk 1]
```diff
@@ -2,9 +2,9 @@ INFO: Available configuration flavors:
 INFO: openshift-4.16-agent-based-baremetal:	Red Hat OpenShift Container Platform 4.16 on Baremetal Agent-based Assisted Installation [Experimental]
 INFO: openshift-4.15-agent-based-baremetal:	Red Hat OpenShift Container Platform 4.15 on Baremetal Agent-based Assisted Installation [Experimental]
 INFO: openshift-4.14-agent-based-baremetal:	Red Hat OpenShift Container Platform 4.14 on Baremetal Agent-based Assisted Installation [Experimental]
+INFO: openshift-4.17-baremetal:	Red Hat OpenShift Container Platform 4.17 on Baremetal
 INFO: openshift-4.16-baremetal:	Red Hat OpenShift Container Platform 4.16 on Baremetal
 INFO: openshift-4.15-baremetal:	Red Hat OpenShift Container Platform 4.15 on Baremetal
-INFO: openshift-4.14-baremetal:	Red Hat OpenShift Container Platform 4.14 on Baremetal
 INFO: openshift-4.13-baremetal:	Red Hat OpenShift Container Platform 4.13 on Baremetal
 INFO: openshift-4.12-baremetal:	Red Hat OpenShift Container Platform 4.12 on Baremetal
 INFO: openshift-4.11-baremetal:	Red Hat OpenShift Container Platform 4.11 on Baremetal
```
Comment (snaiksat) [inline provision/testdata/list_flavors.stdout.txt:10]: only 4.17, 4.16, 4.15 should be visible

File: provision/testdata/list_flavors.stdout.txt [hunk 2]
```diff
@@ -13,9 +13,9 @@ INFO: openshift-4.9-baremetal:	Red Hat OpenShift Container Platform 4.9 on Barem
 INFO: openshift-4.8-baremetal:	Red Hat OpenShift Container Platform 4.8 on Baremetal
 INFO: openshift-4.7-baremetal:	Red Hat OpenShift Container Platform 4.7 on Baremetal
 INFO: openshift-4.6-baremetal:	Red Hat OpenShift Container Platform 4.6 on Baremetal
+INFO: openshift-4.17-openstack:	Red Hat OpenShift Container Platform 4.17 on OpenStack
 INFO: openshift-4.16-openstack:	Red Hat OpenShift Container Platform 4.16 on OpenStack
 INFO: openshift-4.15-openstack:	Red Hat OpenShift Container Platform 4.15 on OpenStack
-INFO: openshift-4.14-openstack:	Red Hat OpenShift Container Platform 4.14 on OpenStack
 INFO: openshift-4.13-openstack:	Red Hat OpenShift Container Platform 4.13 on OpenStack
 INFO: openshift-4.12-openstack:	Red Hat OpenShift Container Platform 4.12 on OpenStack
 INFO: openshift-4.11-openstack:	Red Hat OpenShift Container Platform 4.11 on OpenStack
```
Comment (snaiksat) [inline provision/testdata/list_flavors.stdout.txt:21]: only 4.17, 4.16, 4.15 should be visible

File: provision/testdata/list_flavors.stdout.txt [hunk 3]
```diff
@@ -27,9 +27,9 @@ INFO: openshift-4.6-openstack:	Red Hat OpenShift Container Platform 4.6 on OpenS
 INFO: openshift-4.16-agent-based-esx:	Red Hat OpenShift Container Platform 4.16 on ESX Agent-based Assisted Installation [Experimental]
 INFO: openshift-4.15-agent-based-esx:	Red Hat OpenShift Container Platform 4.15 on ESX Agent-based Assisted Installation [Experimental]
 INFO: openshift-4.14-agent-based-esx:	Red Hat OpenShift Container Platform 4.14 on ESX Agent-based Assisted Installation [Experimental]
```
Comment (snaiksat) [inline provision/testdata/list_flavors.stdout.txt:29]: only 4.16, 4.15 should be visible

File: provision/testdata/list_flavors.stdout.txt [hunk 4]
```diff
@@ -27,9 +27,9 @@ INFO: openshift-4.6-openstack:	Red Hat OpenShift Container Platform 4.6 on OpenS
 INFO: openshift-4.16-agent-based-esx:	Red Hat OpenShift Container Platform 4.16 on ESX Agent-based Assisted Installation [Experimental]
```
Comment (snaiksat) [inline provision/testdata/list_flavors.stdout.txt:27]: shouldn't we be adding openshift-4.17-agent-based-esx flavor as well?

**Review by snaiksat — CHANGES_REQUESTED** [review]

**Review by abhis2112 — APPROVED** [review]

**Review by snaiksat — APPROVED** [review]

**Pull request merged**